### ISSUE TITLE
Run command depending on user

### ISSUE THREAD

**Issue by rambip** [apognu/tuigreet#42]
I would like to start a graphical session (gnome, sway, xmonad, and so on) depending on the user.
I tried to do 
```--cmd 'sh -c \$START_SESSION_SCRIPT'
```
But that didn't work.

Since there is no standard way to start a graphical session (except startx but this was created for X), I don't know how to do that.

I know this is not a good issue, but help would be really apreciated.

**Comment by rambip** [comment]
Thanks, this is really helpfull !
And with nixos I will be able to create the script in my global config, this could be a really nice solution !

Just for curiosity sake, would that work if the content of my file is as follows:

```
!/bin/sh

source $HOME/.profile
exec $START_SESSION_COMMAND
# START_SESSION_COMMAND is defined in the user config
```

**Comment by apognu** [comment]
Absolutely, the session is executed _by_ your user, so `$HOME` will be defined and you can start a session exactly like that.

What `$START_SESSION_COMMAND` should do will depend on the program you need to run, obviously, but I'm sure you already have that lying around.

**Comment by apognu** [comment]
@rambip Did that answer your question?

**Comment by rambip** [comment]
Yeah thanks a lot !
I just need to dig a bit to know when exactly the user environment variables are set on nixos, and it will be good.
